### ISSUE TITLE
Can no longer load mods after updating to the latest version of `collective`

### ISSUE THREAD

**Issue by ChristopherHaws** [ricksouth/serilum-mc-mods#1508]
## **Information**
Minecraft version: 1.19.2
Modloader: Fabric
Fabric loader version: 0.14.12
Environment: Server Startup

Mod name: Cycle Paintings, Inventory Totem, Set World Spawn Point, Stack Refill, Superflat, Vote Command, and I am sure more
Mod version: All of them are the latest versions for 1.19.2

## **Description**
After updating to the latest version of `collective-1.19.2-6.1` I can no longer load other mods that depend on it. Based on the logs it seems like the modid of collective changed from `collective-fabric` to just `collective`, but you would probably know better than I! :)

## **Crash report**
```log
[19:07:14] [main/WARN]: Mod resolution failed
[19:07:14] [main/INFO]: Immediate reason: [HARD_DEP_NO_CANDIDATE cyclepaintings-fabric 2.8 {depends collective-fabric @ [>=5.3]}, ROOT_FORCELOAD_SINGLE cyclepaintings-fabric 2.8]
[19:07:14] [main/INFO]: Reason: [HARD_DEP cyclepaintings-fabric 2.8 {depends collective-fabric @ [>=5.3]}, HARD_DEP inventorytotem-fabric 2.2 {depends collective-fabric @ [>=5.19]}, HARD_DEP setworldspawnpoint-fabric 2.7 {depends collective-fabric @ [>=5.14]}, HARD_DEP stackrefill-fabric 3.2 {depends collective-fabric @ [>=5.4]}, HARD_DEP superflatworldnoslimes-fabric 2.2 {depends collective-fabric @ [>=5.3]}, HARD_DEP votecommand-fabric 2.3 {depends collective-fabric @ [>=5.3]}]
[19:07:14] [main/INFO]: CLIENT environment disabled: [sdlink]
[19:07:14] [main/INFO]: Fix: add [add:collective-fabric 5.19 ([[5.19,?)])], remove [], replace []
[19:07:14] [main/ERROR]: Incompatible mod set!
net.fabricmc.loader.impl.FormattedException: Mod resolution encountered an incompatible mod set!
A potential solution has been determined:
	 - Install collective-fabric, version 5.19 or later.
Unmet dependency listing:
	 - Mod 'Cycle Paintings (Fabric)' (cyclepaintings-fabric) 2.8 requires version 5.3 or later of collective-fabric, which is missing!
	 - Mod 'Inventory Totem (Fabric)' (inventorytotem-fabric) 2.2 requires version 5.19 or later of collective-fabric, which is missing!
	 - Mod 'Set World Spawn Point (Fabric)' (setworldspawnpoint-fabric) 2.7 requires version 5.14 or later of collective-fabric, which is missing!
	 - Mod 'Stack Refill (Fabric)' (stackrefill-fabric) 3.2 requires version 5.4 or later of collective-fabric, which is missing!
	 - Mod 'Superflat World No Slimes (Fabric)' (superflatworldnoslimes-fabric) 2.2 requires version 5.3 or later of collective-fabric, which is missing!
	 - Mod 'Vote Command (Fabric)' (votecommand-fabric) 2.3 requires version 5.3 or later of collective-fabric, which is missing!
	at net.fabricmc.loader.impl.FabricLoaderImpl.load(FabricLoaderImpl.java:190) ~[fabric-loader-0.14.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.init(Knot.java:148) ~[fabric-loader-0.14.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.Knot.launch(Knot.java:68) [fabric-loader-0.14.12.jar:?]
	at net.fabricmc.loader.impl.launch.knot.KnotClient.main(KnotClient.java:23) [fabric-loader-0.14.12.jar:?]
	at org.prismlauncher.launcher.impl.StandardLauncher.launch(StandardLauncher.java:88) [NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.listen(EntryPoint.java:126) [NewLaunch.jar:?]
	at org.prismlauncher.EntryPoint.main(EntryPoint.java:71) [NewLaunch.jar:?]
```

[Full logs](https://mclo.gs/1pLVFyk)

**Comment by ricksouth** [comment]
Hi, thanks for the interest in my mods!

I'm currently in the middle of moving my mod infrastructure to all use the same Jar file. It's going to take me about a week I think.

I've made it so that old Forge mods are compatible with the new format. For old format Fabric mods, you'll need to use Collective-Fabric. The CurseForge app should do this automatically.

TL;DR: Use https://curseforge.com/minecraft/mc-mods/collective-fabric until about next week when everything has been updated :)

**Comment by ricksouth** [comment]
For 1.19.2 that's this file:
https://curseforge.com/minecraft/mc-mods/collective-fabric/files/4136183

**Comment by James103** [comment]
> I'm currently in the middle of moving my mod infrastructure to all use the same Jar file.

Are you trying to make your mods be like Distant Horizons (in terms of loader support)? From their FAQ:
> Q: Forge or Fabric?
> A: Both in the same jar.

**Comment by ricksouth** [comment]
> > I'm currently in the middle of moving my mod infrastructure to all use the same Jar file.
> 
> Are you trying to make your mods be like Distant Horizons (in terms of loader support)? From their FAQ:
> > Q: Forge or Fabric?
> > A: Both in the same jar.

Haven't played their mods before, but I guess so! I'm moving all my projects to Multiloader, and basically splitting the common source, exporting them twice and merging the code. The file size will be a bit bigger, but because only text is duplicated I think it's worth it.

One of the main reasons is to have only one project page per mod, but still have a clean file list. I didn't like the idea of updating the Forge version 3 times, for 3 supported Minecraft versions, and having the latest Fabric file be in the 10th spot.

Might be a few bugs, but it should all work flawlessly very soon 😄

**Comment by ricksouth** [comment]
@ChristopherHaws I think I might be able to add temporary support for the older Fabric mods by adding Collective-Fabric as a JiJ. I'll try this in a few hours and will keep you updated!

**Comment by ricksouth** [comment]
Collective version 6.4 is compatible with the non-updated Fabric mods. So you're good to go on using it! I'll slowly upload the others in the upcoming days. Thanks for opening the issue and giving the mods a try!

https://www.curseforge.com/minecraft/mc-mods/collective/files
